File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.java
class Solution {
    public int kthSmallest(int[][] matrix, int k) {
        int n = matrix.length;
        int m = matrix[0].length;
        PriorityQueue<Integer> pq = new PriorityQueue<>();
        for (int i = 0; i < n; ++i)
            for (int j = 0; j < m; ++j)
                pq.offer(matrix[i][j]);

        while (--k > 0 && pq.size() > 0)
            pq.poll();
        return pq.peek();
    }
}

File: 0378-kth-smallest-element-in-a-sorted-matrix/0378-kth-smallest-element-in-a-sorted-matrix.cs
public class Solution {
    public int KthSmallest(int[][] matrix, int k) {
        int yLen = matrix.Length;
        PriorityQueue<(int val, int row, int col), int> pq = new();

        for (int i = 0; i < yLen; i++)
            pq.Enqueue((matrix[i][0], i, 0), matrix[i][0]); 

        for (int i = 0; i < k - 1; i++) {
            var (val, row, col) = pq.Dequeue();
            if (++col < yLen)
                pq.Enqueue((matrix[row][col], row, col),matrix[row][col]);
        }

        return pq.Dequeue().val;
    }
}